api/lib/domain/usecases/update-last-question-state.js

```diff
@@ -0,0 +1,39 @@
+const logger = require('../../infrastructure/logger');
+const Assessment = require('../models/Assessment');
+
+module.exports = async function updateLastQuestionState({
+  assessmentId,
+  lastQuestionState,
+  challengeId,
+  domainTransaction,
+  assessmentRepository,
+  challengeRepository,
+}) {
+  if (lastQuestionState === Assessment.statesOfLastQuestion.FOCUSEDOUT && challengeId !== undefined) {
```

Comment:
Suggestion: si le `challengeId` ne doit pas être undefined (= aucune situation front connue), je propose un early exit avec un log approprié.
```suggestion
  if(_.isUndefined(challengeId)){
     logger.warn('An event has been received with no challengeId');
     return null;
  }
  if (lastQuestionState === Assessment.statesOfLastQuestion.FOCUSEDOUT) {
```

Reply:
Le `challengeId` [**peut** être `undefined`](https://github.com/1024pix/pix/pull/4239/files#diff-284b0d5ed17e0ba1c6d6b98842b2e0885ab7477bf139bd805166b711b284a375R8). C'est un cas où on veut bien faire l'update (`state` non `focusedout` et/ou juste pas de `challengeId`) donc pas d'early return ;)

Pour le `return null`, effectivement ça me parait bizarre !

Reply:
>Le challengeId peut être undefined
:thinking: Fonctionellement, ça correspond à quoi un évènement front hors d'une question ?

Reply:
On en discute à l'oral